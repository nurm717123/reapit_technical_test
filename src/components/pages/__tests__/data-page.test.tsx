import React from 'react'
import { render } from '@testing-library/react'
import DataPage from '../data/data-page'
import { QueryClient, QueryClientProvider } from 'react-query'

describe('DataPage', () => {
  it('should match a snapshot', () => {
    const queryClient = new QueryClient({
      defaultOptions: {
        // queries: {
        //   refetchOnMount: process.env.NODE_ENV === 'production',
        //   refetchOnWindowFocus: process.env.NODE_ENV === 'production',
        // },
      },
    })
    expect(
      render(
        <QueryClientProvider client={queryClient}>
          <DataPage />
        </QueryClientProvider>,
      ),
    ).toMatchSnapshot()
  })
})
