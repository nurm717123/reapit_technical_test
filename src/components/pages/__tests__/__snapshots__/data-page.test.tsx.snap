// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataPage should match a snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <mock-styled-div>
        <mock-styled-aside>
          <mock-styled-h1>
            Data
          </mock-styled-h1>
          <mock-styled-span>
            <svg
              fill="none"
              height="1em"
              viewBox="0 0 80 80"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M65 35.4285L38 26L11 35.4285L38 45L65 35.4285Z"
                fill="#23A4DE"
              />
              <path
                d="M65 35.4285L38 26V45L65 35.4285Z"
                fill="#0061A8"
              />
              <path
                d="M50.564 12.2517C49.9575 5.9453 44.676 1 38.2717 1C31.8676 1 26.5864 5.94539 25.9794 12.2517C21.8788 12.8404 18.7161 16.4077 18.7161 20.7066C18.7161 25.4157 22.5117 29.2468 27.177 29.2468C53.1412 29.2468 25.0325 29.2469 49.3663 29.2469C54.0317 29.2469 57.8272 25.4157 57.8272 20.7067C57.8272 16.4077 54.6644 12.8404 50.564 12.2517Z"
                fill="#FFB71B"
              />
              <path
                d="M45.8486 24.4714C45.7921 24.3332 45.6601 24.2429 45.5137 24.2429H41.5415L41.5415 15C39.5391 15 37.7646 15 35.0166 15L35.0166 24.243H31.0291C30.8827 24.243 30.7507 24.3332 30.6942 24.4707C30.6383 24.609 30.6688 24.768 30.7725 24.8737L38.0042 32.274C38.0724 32.3435 38.1645 32.3827 38.2609 32.3827C38.3573 32.3827 38.4494 32.3435 38.5175 32.2747L45.7703 24.8745C45.874 24.7687 45.9052 24.6097 45.8486 24.4714Z"
                fill="white"
              />
              <path
                d="M38.2717 34.4135C35.497 34.4135 33.2329 32.1083 33.1607 29.2468H25.2345C25.2822 29.6934 25.3475 30.1374 25.4336 30.5757C25.5387 31.1109 25.9923 31.4954 26.5187 31.4954H28.5213C28.5792 31.6502 28.6406 31.8036 28.7053 31.9555L27.2889 33.4253C26.9166 33.8115 26.8579 34.4164 27.1485 34.872C27.6823 35.709 28.3012 36.4947 28.9881 37.2074C29.6757 37.9209 30.4329 38.5631 31.239 39.1163C31.6779 39.4178 32.2607 39.3566 32.6329 38.9704L34.0493 37.5008C34.1957 37.568 34.3436 37.6317 34.4929 37.6918V39.7697C34.4929 40.3159 34.8634 40.7866 35.3792 40.8957C36.3262 41.0959 37.2993 41.1974 38.2717 41.1974C39.244 41.1974 40.2171 41.0959 41.1642 40.8957C41.6801 40.7866 42.0506 40.3159 42.0506 39.7697L42.0505 37.6918C42.1998 37.6317 42.3477 37.568 42.4941 37.5008L43.9101 38.9702C44.2823 39.3563 44.865 39.4174 45.304 39.1162C46.1098 38.5633 46.8672 37.9209 47.5552 37.2071C48.2427 36.4936 48.8615 35.7079 49.3947 34.8716C49.6851 34.4161 49.6264 33.8114 49.2542 33.4251L47.838 31.9557C47.9028 31.8038 47.9642 31.6503 48.0221 31.4954H50.0243C50.5507 31.4954 51.0043 31.1109 51.1094 30.5756C51.1955 30.1373 51.2609 29.6934 51.3086 29.2468H43.3828C43.3107 32.1083 41.0465 34.4135 38.2717 34.4135Z"
                fill="white"
              />
              <path
                d="M38 44.566L11 35V69.401L38 78V44.566Z"
                fill="#31448F"
              />
              <path
                d="M65 69.401V35L38 44.566V78L65 69.401Z"
                fill="#262F69"
              />
              <path
                d="M11.2576 35L2 46.0304L29.2457 56L38 44.4747L11.2576 35Z"
                fill="#7BC9EB"
              />
              <path
                d="M64.7401 35L74 46.0304L46.7566 56L38 44.4747L64.7401 35Z"
                fill="#23A4DE"
              />
            </svg>
          </mock-styled-span>
          <mock-styled-h2>
            Data Fetching
          </mock-styled-h2>
          <mock-styled-p>
            This simple example demonstrates how to fetch data from our Appointment Config service, authenticated with Reapit Connect using the Connect Session library. You can view the relevant docs below.
          </mock-styled-p>
          <mock-styled-button>
            Connect Session
          </mock-styled-button>
          <mock-styled-button>
            REST API
          </mock-styled-button>
        </mock-styled-aside>
        <mock-styled-section
          classname="el-hfull"
        >
          <mock-styled-h1>
            Properties for Sale
          </mock-styled-h1>
          <mock-styled-div
            classname=""
          >
            <mock-styled-p>
              loading
            </mock-styled-p>
            <mock-styled-div>
              <mock-styled-div />
            </mock-styled-div>
          </mock-styled-div>
        </mock-styled-section>
        <mock-styled-div>
          <div
            class="el-pn-icon"
            data-testid="close-icon"
          >
            <mock-styled-span>
              <svg
                fill="none"
                height="1em"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.99958 0C4.47746 0 0 4.47746 0 9.99958C0 15.5217 4.47746 20 9.99958 20C15.5217 20 20 15.5217 20 9.99958C20 4.47746 15.5217 0 9.99958 0ZM12.0813 15.498C11.5666 15.7012 11.1568 15.8552 10.8495 15.9619C10.5431 16.0686 10.1867 16.1219 9.78116 16.1219C9.1581 16.1219 8.67302 15.9695 8.32762 15.6656C7.98222 15.3617 7.81037 14.9765 7.81037 14.5084C7.81037 14.3263 7.82307 14.1401 7.84847 13.9505C7.87471 13.7608 7.91619 13.5475 7.97291 13.3079L8.61714 11.0324C8.67386 10.814 8.72296 10.6066 8.7619 10.4135C8.80085 10.2188 8.81947 10.0402 8.81947 9.87767C8.81947 9.58815 8.75937 9.38497 8.64 9.27069C8.51894 9.1564 8.29122 9.10053 7.95175 9.10053C7.78582 9.10053 7.61481 9.12508 7.43958 9.17672C7.26603 9.23005 7.11534 9.27831 6.99175 9.32571L7.1619 8.62476C7.58349 8.45291 7.9873 8.30561 8.37249 8.1837C8.75767 8.06011 9.12169 7.99915 9.46455 7.99915C10.0834 7.99915 10.5608 8.14984 10.8969 8.44783C11.2313 8.74667 11.3998 9.13524 11.3998 9.6127C11.3998 9.71175 11.3879 9.88614 11.3651 10.135C11.3422 10.3848 11.299 10.6125 11.2364 10.8216L10.5956 13.0904C10.5431 13.2724 10.4965 13.4806 10.4542 13.7134C10.4127 13.9462 10.3924 14.124 10.3924 14.2434C10.3924 14.5448 10.4593 14.7505 10.5947 14.8597C10.7285 14.9689 10.963 15.0239 11.2948 15.0239C11.4514 15.0239 11.6267 14.996 11.8248 14.9418C12.0212 14.8876 12.1634 14.8394 12.2531 14.7979L12.0813 15.498ZM11.9678 6.2891C11.669 6.56677 11.3092 6.70561 10.8885 6.70561C10.4686 6.70561 10.1062 6.56677 9.80487 6.2891C9.50519 6.01143 9.35365 5.67365 9.35365 5.27915C9.35365 4.8855 9.50603 4.54688 9.80487 4.26667C10.1062 3.98561 10.4686 3.84593 10.8885 3.84593C11.3092 3.84593 11.6698 3.98561 11.9678 4.26667C12.2667 4.54688 12.4165 4.8855 12.4165 5.27915C12.4165 5.6745 12.2667 6.01143 11.9678 6.2891Z"
                  fill="currentColor"
                />
              </svg>
            </mock-styled-span>
          </div>
          <div
            class="el-pn-content"
          >
            you can edit each properties by click the expand button
          </div>
        </mock-styled-div>
      </mock-styled-div>
    </div>
  </body>,
  "container": <div>
    <mock-styled-div>
      <mock-styled-aside>
        <mock-styled-h1>
          Data
        </mock-styled-h1>
        <mock-styled-span>
          <svg
            fill="none"
            height="1em"
            viewBox="0 0 80 80"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M65 35.4285L38 26L11 35.4285L38 45L65 35.4285Z"
              fill="#23A4DE"
            />
            <path
              d="M65 35.4285L38 26V45L65 35.4285Z"
              fill="#0061A8"
            />
            <path
              d="M50.564 12.2517C49.9575 5.9453 44.676 1 38.2717 1C31.8676 1 26.5864 5.94539 25.9794 12.2517C21.8788 12.8404 18.7161 16.4077 18.7161 20.7066C18.7161 25.4157 22.5117 29.2468 27.177 29.2468C53.1412 29.2468 25.0325 29.2469 49.3663 29.2469C54.0317 29.2469 57.8272 25.4157 57.8272 20.7067C57.8272 16.4077 54.6644 12.8404 50.564 12.2517Z"
              fill="#FFB71B"
            />
            <path
              d="M45.8486 24.4714C45.7921 24.3332 45.6601 24.2429 45.5137 24.2429H41.5415L41.5415 15C39.5391 15 37.7646 15 35.0166 15L35.0166 24.243H31.0291C30.8827 24.243 30.7507 24.3332 30.6942 24.4707C30.6383 24.609 30.6688 24.768 30.7725 24.8737L38.0042 32.274C38.0724 32.3435 38.1645 32.3827 38.2609 32.3827C38.3573 32.3827 38.4494 32.3435 38.5175 32.2747L45.7703 24.8745C45.874 24.7687 45.9052 24.6097 45.8486 24.4714Z"
              fill="white"
            />
            <path
              d="M38.2717 34.4135C35.497 34.4135 33.2329 32.1083 33.1607 29.2468H25.2345C25.2822 29.6934 25.3475 30.1374 25.4336 30.5757C25.5387 31.1109 25.9923 31.4954 26.5187 31.4954H28.5213C28.5792 31.6502 28.6406 31.8036 28.7053 31.9555L27.2889 33.4253C26.9166 33.8115 26.8579 34.4164 27.1485 34.872C27.6823 35.709 28.3012 36.4947 28.9881 37.2074C29.6757 37.9209 30.4329 38.5631 31.239 39.1163C31.6779 39.4178 32.2607 39.3566 32.6329 38.9704L34.0493 37.5008C34.1957 37.568 34.3436 37.6317 34.4929 37.6918V39.7697C34.4929 40.3159 34.8634 40.7866 35.3792 40.8957C36.3262 41.0959 37.2993 41.1974 38.2717 41.1974C39.244 41.1974 40.2171 41.0959 41.1642 40.8957C41.6801 40.7866 42.0506 40.3159 42.0506 39.7697L42.0505 37.6918C42.1998 37.6317 42.3477 37.568 42.4941 37.5008L43.9101 38.9702C44.2823 39.3563 44.865 39.4174 45.304 39.1162C46.1098 38.5633 46.8672 37.9209 47.5552 37.2071C48.2427 36.4936 48.8615 35.7079 49.3947 34.8716C49.6851 34.4161 49.6264 33.8114 49.2542 33.4251L47.838 31.9557C47.9028 31.8038 47.9642 31.6503 48.0221 31.4954H50.0243C50.5507 31.4954 51.0043 31.1109 51.1094 30.5756C51.1955 30.1373 51.2609 29.6934 51.3086 29.2468H43.3828C43.3107 32.1083 41.0465 34.4135 38.2717 34.4135Z"
              fill="white"
            />
            <path
              d="M38 44.566L11 35V69.401L38 78V44.566Z"
              fill="#31448F"
            />
            <path
              d="M65 69.401V35L38 44.566V78L65 69.401Z"
              fill="#262F69"
            />
            <path
              d="M11.2576 35L2 46.0304L29.2457 56L38 44.4747L11.2576 35Z"
              fill="#7BC9EB"
            />
            <path
              d="M64.7401 35L74 46.0304L46.7566 56L38 44.4747L64.7401 35Z"
              fill="#23A4DE"
            />
          </svg>
        </mock-styled-span>
        <mock-styled-h2>
          Data Fetching
        </mock-styled-h2>
        <mock-styled-p>
          This simple example demonstrates how to fetch data from our Appointment Config service, authenticated with Reapit Connect using the Connect Session library. You can view the relevant docs below.
        </mock-styled-p>
        <mock-styled-button>
          Connect Session
        </mock-styled-button>
        <mock-styled-button>
          REST API
        </mock-styled-button>
      </mock-styled-aside>
      <mock-styled-section
        classname="el-hfull"
      >
        <mock-styled-h1>
          Properties for Sale
        </mock-styled-h1>
        <mock-styled-div
          classname=""
        >
          <mock-styled-p>
            loading
          </mock-styled-p>
          <mock-styled-div>
            <mock-styled-div />
          </mock-styled-div>
        </mock-styled-div>
      </mock-styled-section>
      <mock-styled-div>
        <div
          class="el-pn-icon"
          data-testid="close-icon"
        >
          <mock-styled-span>
            <svg
              fill="none"
              height="1em"
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.99958 0C4.47746 0 0 4.47746 0 9.99958C0 15.5217 4.47746 20 9.99958 20C15.5217 20 20 15.5217 20 9.99958C20 4.47746 15.5217 0 9.99958 0ZM12.0813 15.498C11.5666 15.7012 11.1568 15.8552 10.8495 15.9619C10.5431 16.0686 10.1867 16.1219 9.78116 16.1219C9.1581 16.1219 8.67302 15.9695 8.32762 15.6656C7.98222 15.3617 7.81037 14.9765 7.81037 14.5084C7.81037 14.3263 7.82307 14.1401 7.84847 13.9505C7.87471 13.7608 7.91619 13.5475 7.97291 13.3079L8.61714 11.0324C8.67386 10.814 8.72296 10.6066 8.7619 10.4135C8.80085 10.2188 8.81947 10.0402 8.81947 9.87767C8.81947 9.58815 8.75937 9.38497 8.64 9.27069C8.51894 9.1564 8.29122 9.10053 7.95175 9.10053C7.78582 9.10053 7.61481 9.12508 7.43958 9.17672C7.26603 9.23005 7.11534 9.27831 6.99175 9.32571L7.1619 8.62476C7.58349 8.45291 7.9873 8.30561 8.37249 8.1837C8.75767 8.06011 9.12169 7.99915 9.46455 7.99915C10.0834 7.99915 10.5608 8.14984 10.8969 8.44783C11.2313 8.74667 11.3998 9.13524 11.3998 9.6127C11.3998 9.71175 11.3879 9.88614 11.3651 10.135C11.3422 10.3848 11.299 10.6125 11.2364 10.8216L10.5956 13.0904C10.5431 13.2724 10.4965 13.4806 10.4542 13.7134C10.4127 13.9462 10.3924 14.124 10.3924 14.2434C10.3924 14.5448 10.4593 14.7505 10.5947 14.8597C10.7285 14.9689 10.963 15.0239 11.2948 15.0239C11.4514 15.0239 11.6267 14.996 11.8248 14.9418C12.0212 14.8876 12.1634 14.8394 12.2531 14.7979L12.0813 15.498ZM11.9678 6.2891C11.669 6.56677 11.3092 6.70561 10.8885 6.70561C10.4686 6.70561 10.1062 6.56677 9.80487 6.2891C9.50519 6.01143 9.35365 5.67365 9.35365 5.27915C9.35365 4.8855 9.50603 4.54688 9.80487 4.26667C10.1062 3.98561 10.4686 3.84593 10.8885 3.84593C11.3092 3.84593 11.6698 3.98561 11.9678 4.26667C12.2667 4.54688 12.4165 4.8855 12.4165 5.27915C12.4165 5.6745 12.2667 6.01143 11.9678 6.2891Z"
                fill="currentColor"
              />
            </svg>
          </mock-styled-span>
        </div>
        <div
          class="el-pn-content"
        >
          you can edit each properties by click the expand button
        </div>
      </mock-styled-div>
    </mock-styled-div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
